epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelStaggeredGrid.java

```diff
@@ -0,0 +1,60 @@
+package com.airbnb.epoxy;
+
+import android.view.View;
+import android.view.ViewGroup.LayoutParams;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.StaggeredGridLayoutManager;
+
+/**
+ * An base {@link EpoxyModel} allow you set a model view within StaggeredGridLayoutManager is a
+ * full span model or not.
+ *
+ * @param <T> The view of model view
+ */
+public abstract class EpoxyModelStaggeredGrid<T extends View> extends EpoxyModel<T> {
+
+  /**
+   * The flag to mark a model view is full span or not.
+   */
+  private boolean isFullSpanModel = false;
+
+  /**
+   * The method to set a model view can be full span or not.
+   *
+   * @param isFullSpanModel if true, the model view will be full span.
+   * @return The model view.
+   */
+  public EpoxyModelStaggeredGrid<T> setFullSpanModel(boolean isFullSpanModel) {
```

Comment:
I think this needs to be marked as a model prop, otherwise toggling it will not trigger a view update